src/lib/EditMessage.svelte

```diff
@@ -214,6 +225,56 @@
     document.body.removeChild(a)
   }
 
+  const preprocessMath = (text: string): string => {
+    if (renderLatexFlag !== true) {
+      return text
+    }
+    let codeBlockPlaceholderPrefix = '__prefix__c0d3b10ck__'
+    while (text.indexOf(codeBlockPlaceholderPrefix) > 0) {
+      codeBlockPlaceholderPrefix = codeBlockPlaceholderPrefix + '_'
+    }
+    let index = 0
+    const codeBlocks = []
+
+    const codeBlockRegex = /(```[\s\S]*?```|`[^`]*`)/g
+  
+    text = text.replace(codeBlockRegex, (match) => {
+      const placeholder = `${codeBlockPlaceholderPrefix}idx${index}__`
+      codeBlocks.push(match)
+      index++
+      return placeholder
+    })
+
+    text = text
+      .replace(/(\\\[((?:\s|\S)*?)\\\])|(\$\$((?:\s|\S)*?)\$\$)/g, (match, p1, p2, p3, p4) => {
```

Comment:
this looks like something that is brittle and can break lots of other cases. Is there not a better way to detect latex?

Reply:
I thought so, but looking at KaTeX's own AutoRender, they do does delimeter detecting simmilarly.. so I thought using simple regex is fine. 

I agree with that it might break easily and there should be more solid way, but as far as I see, almost every LaTeX detector uses simmilar regex.

Reply:
I disabled rendering at default and added button to render afterwards if needed.